depthcharge/data/parsers.py

```diff
@@ -82,74 +105,114 @@ def parse_spectrum(self, spectrum: dict) -> MassSpectrum | None:
             The parsed mass spectrum or None if it is skipped.
         """
 
-    def read(self) -> BaseParser:
-        """Read the ms data file.
+    def parse_custom_fields(self, spectrum: dict) -> dict[str, Any]:
```

Comment:
It is not totally clear to me what this is doing ...

Reply:
I'll write up documentation on how to use the `custom_fields` argument, but here's a brief explanation:

Each spectrum parsed by Pyteomics yields a nested dictionary. This function lets users extract values from this nested dictionary by defining the order of elements / indices they need to access a value of interest.

For example, if our spectrum dictionary is `spectrum`, we could access ion mobility as:
```python
{"ion_mobility": spectrum["precursorList"]["precursor"][0]["selectedIonList"]["selectedIon"][0]["inverse reduced ion mobility"]}
```

This function allows users to specify the accession path as a iterable, so that the following would be the equivalent using this function:
```python
{"ion_mobility": ["precursorList", "precursor", 0, "selectedIonList", "selectedIon", 0, "inverse reduced ion mobility"]}
```
Does that make sense?

Reply:
I've changed this to work in a way that I think is more intuitive 😄